### DISCUSSION TITLE
Importing defaults doesn't work under ESM

### DISCUSSION THREAD

**Discussion by magyargergo** [TypeStrong/ts-node#1636] [Q&A] [answered]
Hi there, I'm trying to import a module like this `import * as prettier from 'prettier';` under `NodeJS 16.13.2` with `ts-node: 10.5.0`. I'm using this command `node --experimental-specifier-resolution=node --loader ts-node/esm script.ts` to run my script but I get this error `prettier.format is not a function`. The following image shows an example:

![image](https://user-images.githubusercontent.com/11230420/153380827-cb41a67b-b8fc-4237-8e95-ba020f4fd198.png)

I put the `default` after the module name and the `format` function becomes available.
![image](https://user-images.githubusercontent.com/11230420/153380909-f739da1b-3618-4f11-83e7-675e9e929b5b.png)

This is the snippet of my `tsconfig.json`

```json
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "src",
    "declaration": false,
    "downlevelIteration": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "experimentalDecorators": true,
    "forceConsistentCasingInFileNames": true,
    "importHelpers": true,
    "lib": [
      "es2020",
      "dom"
    ],
    "module": "es2020",
    "moduleResolution": "node",
    "outDir": "./dist",
    "sourceMap": true,
    "strict": true,
    "target": "es2020",
}
```
The `package.json` has `"type": "module"` set.

I found this in the NodeJS documentation: https://nodejs.org/api/esm.html#esm_interoperability_with_commonjs

**Comment by cspotcode** [comment, marked as answer]
You might be hitting this issue, copied from the nodejs documentation that you linked:

> When importing [CommonJS modules](https://nodejs.org/api/esm.html#commonjs-namespaces), the module.exports object is provided as the default export. Named exports may be available, provided by static analysis as a convenience for better ecosystem compatibility.

You are doing a wildcard import, but a default import.  So that's the first thing I would test.
